velox/type/DecimalUtil.h

```diff
@@ -258,17 +258,31 @@ class DecimalUtil {
     // LONG_DOUBLE_MAX.
     long double scaledValue = std::round(
         (long double)value * DecimalUtil::kPowersOfTen[fractionDigits]);
-    if (scale > fractionDigits) {
-      scaledValue *= DecimalUtil::kPowersOfTen[scale - fractionDigits];
-    } else {
-      scaledValue /= DecimalUtil::kPowersOfTen[fractionDigits - scale];
-    }
-
-    const auto result = folly::tryTo<TOutput>(std::round(scaledValue));
+    const auto result = folly::tryTo<TOutput>(scaledValue);
```

Comment:
If we convert `scaledValue` as target type before rescaling, will unexpected overflow happen? For example, a value fits in int64_t after dividing the rescale factor but overflows before the rescaling.

Reply:
Taking double as an example, it allows up to 17 digits, while int64_t allows 19 digits. So, I think this is safe and will not cause an overflow.

Reply:
> long double scaledValue = std::round(
        (long double)value * DecimalUtil::kPowersOfTen[fractionDigits]);
 
If `value` is `kMaxDoubleBelowInt64Max`, after multiplying 10^15, the result will overflow for int64_t, no?

Reply:
I think in this case the fractionDigits is 0.
`const auto fractionDigits = std::max(digits - integralDigits, 0);`

Reply:
a test case for cast (kMaxDoubleBelowInt64Max as short decimal) is added.